extra/genentries

```diff
@@ -187,26 +227,62 @@ function readdesktopfile {
 
         # If we can't find anything that matches the theme, just search anywhere
         if [ -z "$icon" ]; then
-          icon="$(find -L "$pixmaps" -type f \( "${findargs[@]}" -false \) -print -quit)"
+          icon="$(find -L "$pixmaps" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)"
         fi
         if [ -z "$icon" ] && [ -d "$user_home/.local/share/icons" ]; then
-          icon=$(find -L "$user_home/.local/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit)
+          icon=$(find -L "$user_home/.local/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
         fi
         if [ -z "$icon" ] && [ -d "$user_home/.icons" ]; then
-          icon=$(find -L "$user_home/.icons" -type f \( "${findargs[@]}" -false \) -print -quit)
+          icon=$(find -L "$user_home/.icons" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
         fi
         if [ -z "$icon" ] && [ -d "/usr/share/icons" ]; then
-          icon=$(find -L "/usr/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit)
+          icon=$(find -L "/usr/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
+        fi
+        if [ -z "$icon" ] && [ -d "/usr/share/pixmaps/puppy" ]; then
+          icon=$(find -L "/usr/share/pixmaps/puppy" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
         fi
       fi
 
+      # if the icon is a SVG, xlunch cant show it, so convert to PNG and
+      # put it in /usr/share/xlunch/svgicons/<name>.png
```

Comment:
This path can be changed by a runtime switch, don't name a path here.

Reply:
:+1: